lib/pbench/test/unit/server/test_server_audit.py

```diff
@@ -58,52 +76,86 @@ def make_audits(self, client, create_user) -> str:
-
-    def test_get_bad_keys(self, query_get):
-        response = query_get({"xyzzy": "foo"}, HTTPStatus.BAD_REQUEST)
-        assert response.json == {"message": "Unknown URL query keys: xyzzy"}
-
-    def test_get_all(self, query_get, make_audits):
-        """With no query parameters, we should get all audit records"""
-        expected_user_id = make_audits
+            Audit.create(
+                root=root, status=AuditStatus.FAILURE, reason=AuditReason.PERMISSION
+            )
         response = query_get(expected_status=HTTPStatus.OK)
         audits = response.json
-        assert len(audits) == 4
         assert audits[0]["status"] == "BEGIN"
         assert audits[0]["operation"] == "CREATE"
         assert audits[0]["name"] == "first"
-        assert audits[0]["user_id"] == expected_user_id
+        assert audits[0]["user_id"] == str(create_user.id)
         assert audits[0]["user_name"] == "test"
+        assert audits[0]["object_id"] is None
+        assert audits[0]["object_name"] is None
+        assert audits[0]["object_type"] is None
+        assert audits[0]["user_name"] == "test"
+        assert audits[0]["reason"] is None
         assert audits[1]["status"] == "SUCCESS"
         assert audits[1]["operation"] == "CREATE"
         assert audits[1]["name"] == "first"
-        assert audits[1]["user_id"] == expected_user_id
+        assert audits[1]["user_id"] == str(create_user.id)
         assert audits[1]["user_name"] == "test"
+        assert audits[1]["object_id"] is None
+        assert audits[1]["object_name"] is None
+        assert audits[1]["object_type"] is None
+        assert audits[1]["reason"] is None
         assert audits[2]["status"] == "BEGIN"
         assert audits[2]["operation"] == "UPDATE"
         assert audits[2]["name"] == "second"
         assert audits[2]["user_id"] is None
         assert audits[2]["user_name"] == "fake"
+        assert audits[2]["object_id"] == drb.resource_id
+        assert audits[2]["object_name"] == drb.name
+        assert audits[2]["object_type"] == "DATASET"
+        assert audits[2]["reason"] is None
         assert audits[3]["status"] == "FAILURE"
         assert audits[3]["operation"] == "UPDATE"
         assert audits[3]["name"] == "second"
         assert audits[3]["user_id"] is None
         assert audits[3]["user_name"] == "fake"
+        assert audits[3]["object_id"] == drb.resource_id
+        assert audits[3]["object_name"] == drb.name
+        assert audits[3]["object_type"] == "DATASET"
+        assert audits[3]["reason"] == "PERMISSION"
+
+        # The responses are cached so we can avoid repeating the individual
+        # fields in subsequent tests.
+        TestServerAudit.audits = audits
+        return user_id
+
+    def check_audits(self, actual: list[Audit], expected: list[int]):
+        assert len(actual) == len(expected)
+        ai = 0
+
+        for e in expected:
+            assert actual[ai] == self.audits[e]
+            ai += 1
```

Comment:
Alternately,
```python
        for ai, e in enumerate(expected):
            assert actual[ai] == self.audits[e]
```

Reply:
I knew there was a way to do that ... I couldn't remember `enumerate` at the time.